src/test/java/com/ge/predix/metering/filter/NuregoIT.java

```diff
@@ -16,125 +16,277 @@
+		String response = this.nuregoTemplate.postForObject(oauthURL, serviceRequest, String.class);
+		@SuppressWarnings("unchecked")
+		Map<String, Object> responseMap = new ObjectMapper().readValue(response, Map.class);
+		return responseMap.get(Constants.ACCESS_TOKEN).toString();
     }
+    
+	private String retrieveComponent(String serviceInstanceGuid, String nuregoAccessToken) throws Exception{
+
+		String retrieveComponentURL = this.nuregoURL + Constants.SERVICE_COMPONENT_URL + serviceInstanceGuid;
+		
+		HttpHeaders headers = new HttpHeaders();
+		headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+		headers.add(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+		headers.add(HttpHeaders.AUTHORIZATION, Constants.BEARER + nuregoAccessToken);
+		
+		UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(retrieveComponentURL)
+				.queryParam("provider", "cloud-foundry");
+		
+		HttpEntity<?> entity = new HttpEntity<>(headers);
+		ResponseEntity<String> responseEntity = null;
+		long end = System.currentTimeMillis() + 40000;
+		//Attempts to retrieve component for 20 seconds before it throws an exception.
+		//Component id is not updated immediately in Nurego after service instance creation, there is a delay. Hence trying it for 20 seconds.
+	    while(System.currentTimeMillis() < end) {
+	    		try {
+	    			responseEntity = this.nuregoTemplate.exchange(builder.build().encode().toUri(), HttpMethod.GET, entity ,String.class);
+	    			Thread.sleep(2000); // need to sleep because nurego mapping takes time to update
+	    			break;
+	    		} catch(Exception ex) {
+	    			System.out.println("Attempting to retrieve component ID...");
+	    		}
+	    }
+	    if(responseEntity == null) {
+	    		return null;
+	    }
+	    String responseBody = responseEntity.getBody();
+	    
+	    @SuppressWarnings("unchecked")
+		Map<String,Object> map = new ObjectMapper().readValue(responseBody, Map.class);	    
+	    	return map.get(Constants.ID).toString();
+
+	}
+	
+
+	private Date tomorrow() {
+		final Calendar cal = Calendar.getInstance();
+		cal.add(Calendar.DATE, +1);
+		return cal.getTime();
+	}
+	
+	private Date today() {
+		final Calendar cal = Calendar.getInstance();
+		return cal.getTime();
+	}
+	private ComponentUsage retrieveRawUsage(String accessToken, String cmp_id, boolean acsFlag, String featureId) throws JsonParseException, JsonMappingException, IOException {
+		DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+		String retrieveRawUsage = Constants.NUREGO_USAGE_URL;
+		HttpHeaders headers = new HttpHeaders();
+		headers.add(HttpHeaders.ACCEPT, MediaType.ALL_VALUE);
+		headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+		headers.add(HttpHeaders.AUTHORIZATION, Constants.BEARER +accessToken);
+		UriComponentsBuilder builder;
+		if(acsFlag) {
+			 builder = UriComponentsBuilder.fromUriString(retrieveRawUsage)
```

Comment:
If you wanted to simply the code here slightly, you could use:
String serviceId = acsFlag ? Constants.ACS_SERVICE_ID : Constants.UAA_SERVICE_ID;  

Then you would only have the builder code once.